### PR TITLE
feat(build): add rte header files and find_first_bit function to support

### PR DESCRIPTION
**Pull request by wangzihao3**
building on aarch64

Related to: openebs/openebs#1295

- add the rte header files which is needed by arm64 and is included by rte_ring.h.
- in src/istgt_lu.h, include queue.h in different path when arch is different.

Signed-off-by: wangzihao <wangzihao18@huawei.com>

### CODE REVIEW

**Comment by wangzihao3** [comment]
Hi, @mynktl .PTAL.

File: src/istgt_lu.h [hunk 1]
```diff
@@ -58,8 +58,12 @@
 #endif
 
 #ifdef __linux__
+#if (defined(__aarch64__) || defined(__arm__))
+#include <aarch64-linux-gnu/sys/queue.h>
```
Comment (mynktl) [inline src/istgt_lu.h:62]: On Aarch32, this file location will be different. Let's remove `|| defined(__arm__)`.

File: src/Makefile.in [hunk 2]
```diff
@@ -48,7 +48,9 @@ istgt_header = istgt_ver.h istgt.h istgt_iscsi.h istgt_iscsi_xcopy.h istgt_iscsi
 replication_source = replication.c replication_misc.c ring_mempool.c rte_ring.c data_conn.c
 
 replication_header = replication.h  istgt_integration.h zrepl_prot.h 	\
```
Comment (mynktl) [inline src/Makefile.in:50]: `config.sub, config.guess` are out-dated. `configure.ac`. I got this error `configure: error: cannot guess build type; you must specify one`. Can you update these files? Using automake (automake --add-missing --copy --force-missing) or [config.guess](http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD) [config.sub](http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD)

**Review by mynktl — COMMENTED** [review]
Changes looks good. @Wangzihao18 Can you update the `config.guess/sub`? Thanks

File: src/Makefile.in (hunk 2 above)
Comment (wangzihao3) [reply src/Makefile.in:50]: Sorry, I'm at home now, I will update these files tomorrow. May you try `./configure --enable-replication --build=arm-linux` ? It maybe works.

File: src/istgt_lu.h (hunk 1 above)
Comment (wangzihao3) [reply src/istgt_lu.h:62]: Ok, I will update it.

**Review by mynktl — APPROVED** [review]

File: src/arch/x86/rte_atomic.h [hunk 3]
```diff
@@ -217,9 +217,9 @@ static inline int rte_atomic32_dec_and_test(rte_atomic32_t *v)
 #endif
 
 #ifdef RTE_ARCH_I686
-#include "rte_atomic_32.h"
+#include "arch/x86/rte_atomic_32.h"
```
Comment (pawanpraka1) [inline src/arch/x86/rte_atomic.h:220]: where is this file rte_atomic_32.h? I don't see it in the PR.

Comment (wangzihao3) [reply src/arch/x86/rte_atomic.h:220]: There was no such file in the original project. I thought this file was not needed. I find the file in the DPDK repo. https://github.com/DPDK/dpdk/blob/master/lib/librte_eal/common/include/arch/x86/rte_atomic_32.h. I can upload it if needed

File: src/arch/arm/rte_pause.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+#ifndef _RTE_PAUSE_ARM_H_
+#define _RTE_PAUSE_ARM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "arch/arm/rte_pause_64.h"
```
Comment (pawanpraka1) [inline src/arch/arm/rte_pause.h:12]: what is the need of having rte_pause.h and rte_pause_64.h if functionality wise both are same. We can have one header file only here?

File: src/arch/arm/rte_atomic.h [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2015 RehiveTech. All rights reserved.
+ */
+
+#ifndef _RTE_ATOMIC_ARM_H_
+#define _RTE_ATOMIC_ARM_H_
+
+#include "arch/arm/rte_atomic_64.h"
```
Comment (pawanpraka1) [inline src/arch/arm/rte_atomic.h:8]: same here what is the need of rte_atomic_64.h, we can have one file. Ideally we include rte_atomic.h in the code and here in this header file we decide which file to include. Have #ifdef to decide which one to include based on the architecture (32 bit or 64 bit) if there is a difference in the implementation for both architecture.

File: src/arch/arm/rte_pause.h (hunk 4 above)
Comment (wangzihao3) [reply src/arch/arm/rte_pause.h:12]: If there is no need to support arm32, we can combine two files into one. Actually  there also include rte_pause_32.h， but I remove it.

File: src/arch/arm/rte_atomic.h (hunk 5 above)
Comment (wangzihao3) [reply src/arch/arm/rte_atomic.h:8]: yes, good point

File: src/arch/x86/rte_atomic.h (hunk 3 above)
Comment (pawanpraka1) [reply src/arch/x86/rte_atomic.h:220]: lets raise a separate PR for that.

File: src/arch/arm/rte_pause.h (hunk 4 above)
Comment (mynktl) [reply src/arch/arm/rte_pause.h:12]: Better not to combine two files. Keep both the file separate.

File: src/arch/x86/rte_atomic.h (hunk 3 above)
Comment (wangzihao3) [reply src/arch/x86/rte_atomic.h:220]: ok, I will raise it later

**Review by pawanpraka1 — APPROVED** [review]
changes are fine for 64 bit system.

**Comment by wangzihao3** [comment]
thanks @pawanpraka1 . I raised a PR in libcstor repo to support 32-bit system.  https://github.com/openebs/libcstor/pull/36 .  Thank you very much if you can give me some advice.

**Pull request merged**